File: modules/common-model/src/main/java/com/shterneregen/securelan/common/model/ChatMessage.java
package com.shterneregen.securelan.common.model;

import java.time.Instant;
import java.util.Objects;

public record ChatMessage(
        String id,
        String senderId,
        String recipientId,
        String content,
        Instant createdAt
) {
    public ChatMessage {
        Objects.requireNonNull(id, "id must not be null");
        Objects.requireNonNull(senderId, "senderId must not be null");
        Objects.requireNonNull(recipientId, "recipientId must not be null");
        Objects.requireNonNull(content, "content must not be null");
        Objects.requireNonNull(createdAt, "createdAt must not be null");
        if (content.isBlank()) {
            throw new IllegalArgumentException("content must not be blank");
        }
    }
}

File: modules/common-model/src/main/kotlin/com/shterneregen/securelan/common/model/ChatMessage.kt
package com.shterneregen.securelan.common.model

import java.time.Instant
import java.util.Objects

@JvmRecord
data class ChatMessage(
    val id: String,
    val senderId: String,
    val recipientId: String,
    val content: String,
    val createdAt: Instant,
) {
    init {
        Objects.requireNonNull(id, "id must not be null")
        Objects.requireNonNull(senderId, "senderId must not be null")
        Objects.requireNonNull(recipientId, "recipientId must not be null")
        Objects.requireNonNull(content, "content must not be null")
        Objects.requireNonNull(createdAt, "createdAt must not be null")
        require(content.isNotBlank()) { "content must not be blank" }
    }
}
